src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -207,23 +207,38 @@ def __init__(self, parent, fl, widgetMode):
         self.spatial_extent_box.setOutputExtentFromCurrent()
         self.spatial_extent_box.setMapCanvas(map_canvas)
 
+        self.restore_settings()
```

Comment:
Please use more relevant names when creating new classes/methods/variables etc. This will save everybody's time in six months when we come back and re-read the code.

In this case, `restore_settings()` is incorrect, or at least too vague. The method is restoring search filters, not settings.

```suggestion
        self.restore_search_filters()
```